### PR TITLE
Removes R&D gun production

### PR DESCRIPTION
**Pull request by Iamgoofball**
reasons for this:
1. R&D outclasses security at "killing da antags" rapidly
2. R&D becomes the centerpoint of every gamemode, blob in specific
3. constant arguments over R&D gun balance with nonstop revision and changes of everything else to balance around R&D mass printing guns
4. there's no difficulty aside from waiting 15 minutes to get all the materials you need to mass produce guns
5. constant text walls from R&D mains every time we try to nerf their guns trying to make us think the problem is something completely unrelated to R&D guns
6. R&D is shit

i left the gimmicky shit like portal guns and the gravity gun because those are actually unique

@KorPhaeron

### CODE REVIEW

**Comment by XDTM** [comment]
I actually kinda approve, but i bet people will keep saying RnD is op regardless of any nerfs

**Comment by ChangelingRain** [comment]
if they continue then we know they actually wanted it nerfed so they'll be happy.

**Comment by Iamgoofball** [comment]
additional bonus is we can now balance blob and gang and cult and clockcult and every defensive position game without worrying about xray lasers destroying everything

**Comment by kevinz000** [comment]
Shit change

**Comment by Grazyn** [comment]
Make a thing in sec that is synched to rnd and only makes those guns

**Comment by Iamgoofball** [comment]
feel free to exercise your freedom of speech here @kevinz000 

jobe

**Comment by kevinz000** [comment]
coming from the loudest person near every single headmin election lol

**Comment by Spaceinaba** [comment]
if you're gonna do meme pr's just up the ante and make one that deletes every single line of code

**Comment by Iamgoofball** [comment]
@Spaceinaba please join kevin in the free speech zone with the rest of the R&D mains

**Comment by SgtxRootbeer** [comment]
@Iamgoofball keep the tesla revolver, since its also unique and is science-themed.
also, let science print basic wt-550 autorifle ammo since we cant make it at an autolathe, and ordering more crates of them is shitty.

**Comment by SgtxRootbeer** [comment]
also, with this change in place, we might be able to bring back telescience.

**Comment by optimumtact** [comment]
>bring back telescience

Delusional

**Comment by Bawhoppen** [comment]
Anyone who argues with this change is a science main and their argument is therefore forfeit

**Comment by SgtxRootbeer** [comment]
@bawhoppen I am a science main, and I still like this change, except for the removal of the Tesla gun and the wt-550 ammo

**Comment by arsserpentarium** [comment]
RnD still haven't firing pins. So it's difficult to use all this guns.
Also I think, that Rnd should have more guns. But peaceful ones. Hallucinogun, janitor's gun, maybe someting more strange.

Yes,RnD is no good, but before we have something better, we should keep it fun.

**Comment by PKPenguin321** [comment]
This does not remove R&D

**Comment by Iamgoofball** [comment]
It's funny how people equate taking away R&D's guns with removing R&D

**Comment by Iamgoofball** [comment]
Stuff like the botanist gun and the portal gun are still in.

**Comment by Qbopper** [comment]
likely won't happen because it's another "goof tries to remove a significant feature out of nowhere citing controversy as the reason" pr, though

**Comment by Incomptinence** [comment]
Probably should keep ion rifle still pretty handy for non murder purposes and prevents imagination stunted clams like pkp declaring cyborgs the ultimate perfect life form or whatever.

**Comment by Slignerd** [comment]
@Incomptinence It's probably not getting merged anyways.

File: code/modules/research/designs/weapon_designs.dm [hunk 1]
```diff
@@ -134,26 +93,6 @@
 	build_path = /obj/item/grenade/chem_grenade/adv_release
 	category = list("Weapons")
 
-/datum/design/xray
```
Comment (Shadowlight213) [inline code/modules/research/designs/weapon_designs.dm]: Might wannanleave xray because bleb

Comment (Iamgoofball) [reply code/modules/research/designs/weapon_designs.dm]: Read the comments before posting.

**Comment by YakumoChen** [comment]
Clock cult and war ops and every other 'loud antag battle mode' is already balanced around science being able to make weapons that can stop them.

**Comment by Bawhoppen** [comment]
@YakumoChen They absolutely are not and you damn well know it.

**Comment by KorPhaeron** [comment]
Yakumo raises a fair point

I also still think RD being so fast and the things it produces so cheap compared to mineral rates are what is wrong with with the balance, not the existence of the guns themselves

**Comment by Bawhoppen** [comment]
I think it's ridiculous to say that R&D's gun production has hand-tailored balance to fit in with how our gamemodes play out.
In virtually every case, it has either a minor negligible influence on the round, or a major undue one.

Normal-style antag rounds (cults, traitors, lings, etc): AEGs and other R&D guns clearly provide an unbalanced effect on the round, with science becoming an arms factory, and completely outclassing both antags AND security, while rendering cargo for new weapons obsolete. Same stuff as I've explained before.

Blob: Blobs are primarily fought by hand with welders, with a select few people having the roundstart armory guns or stuff from cargo. Emitters obviously also see widespread play. Science can first AEGs, which are a pretty minor effect on the fight itself compared to just have more normal lasers. Once Xray lasers are researched the entire gamemode is totally derailed and becomes a completely onesided joke.
R&D's primary benefit in blob is actually experimental welders, which I'd say is a fair balancing point for the mode.

Nukeops: Crew typically are using roundstart armory and makeshift weapons, and then supplemented by cargo ordering new weapons. The sheer quantity and effectiveness of AEGs that can be produced (but oddly often aren't) can very well cause balance issues. Even worse than that, is the fact R&D can mass-spam ion carbines which completely throw off any of our existing balance of crew VS armor, and are a huge balance issue.

Clockcult: New clockcult largely involved the same stuff as I mentioned with nukeops, but also genetics, mechs, and other late-round features. Not as experienced in the new mode.


You also claim that our gamemodes have been specifically balanced to properly include R&D, however I'd like you to point me to one PR to where someone specifically changed R&D or gamemode balance so they would synergize well together. 
Because as far as I'm aware, the existing balance, which I'd consider to be not very good, has developed naturally just from people messing with R&D and then it becoming a big intrusive part of rounds. 
These modes (with the possible exception of new clockcult) were made with their own balance in mind originally, before R&D became the absolute arsenal it is nowadays.

**Comment by AnonymousNow** [comment]
We need a means by which we can access these guns through a department like cargo bay or security, with some investiture of time and resources, if it's going to be removed from R&D. Else it's just a funhunt change.

**Comment by YakumoChen** [comment]
just sabotage RnD, it's not difficult

**Comment by JohnMcLaime** [comment]
If everyone in clockcult is too retarded to use their teleport ability to take out rnd and spare boards, then they deserve to lose.

Blob can stop it by deploying near mining/sci (still there are spare boards/gulag shuttle to worry about)

Same with bloodcult, they have veil shifter to quickly get in and get out.

Gangs could easily stop rnd since they have c4 or can just hack in, same with spare board storage.

Revs - nuff said.

Warops are meant to be a challenge, since they are throwing out their greatest asset - gamemode secrecy.

Even then, what "good" guns are there, besides adv eguns which are good due to 1 hit stuns? 
Beamrifle can be literally outhealed due to its shit rate of fire.
Temp rifle is an okay tacticool tool but requires another weapon to supplement it.
Tesla revolver, wasnt it nerfed already? (ie doesnt oneshot kill by exploding machinery nearby)
Xray requires thermals for best use and is mostly anti blob (unless everyone in clockcult is a moron/newbie with no foresight).
Ionguns? Nukies arent forced to buy mechs or borgs yaknow and should plan accordingly.

Then there are firing pins, rnd only starts with 5, which can hardly be called "mass production" and getting more pins is relying on other departments or breaking into them to get pins or mindshield implants.

Rnd is better than sec at killing antags? Sec has basically 1 hit kill guns which are tasers, tase and lase/cuff and raep.

Again, all antags besides warops (nobody forcing them to declare war anyway, it is a > challenge mode) and blob to some extent, have means to curb rnd early into the round.

**Comment by Arianya** [comment]
Its another "goof refuses to believe anyone else's view of game balance for the game he doesn't play" episode.

**Comment by Arianya** [comment]
Also seriously just lock discussion to contributors before this becomes another 100+ comment chain of people who aren't going to change each other's mind

**Comment by pantherpusher** [comment]
What if RnD can only make extremely ineffective prototypes but then you can send one on the supply shuttle and centcom will start producing full functioning versions to be sold to the station?

**Comment by Ratvar** [comment]
gunhog silently liking and disliking things without any comment, aaas always

File: code/modules/research/designs/weapon_designs.dm (hunk 1 above)
Comment (Shadowlight213) [reply code/modules/research/designs/weapon_designs.dm]: Ah. Missed that part.
Still might need to be accompanied by blob nerfs in a followup pr.

**Comment by Shadowlight213** [comment]
This is a good pr.

**Review by Shadowlight213 — APPROVED** [review]

**Comment by sawrge** [comment]
@DoctoreMad there's no winning or losing an argument when either side has adamantly refused to budge for the past n years

**Comment by davethwave** [comment]
There's still techweb coming that will make getting guns harder to do

**Comment by Bawhoppen** [comment]
If you think that's true with Kev at the helm, I have bad news

**Comment by Robustin** [comment]
I think part of the issue is that every part of science is more powerful than the entirety of other departments. Mechs, Borgs, TTVs, Xenobio slave armies - each of those is fucking God-Tier for shaping the outcome of a round and then you've got RND on top of that as the epicenter. Even minus guns RND carries tremendous influence with being able to control AI laws, create Warop-Tier implants, an arsenal of "non-firearm" weaponry (chem grenades, rapid fire syringe guns), top-tier armor and HUDs, every machine board, etc.

Being able to make AND use your own insane weaponry is just icing on the icing on the icing on the icing of the power-creep cake that science has come to expect every round. But its still the only play we can imagine creating "wacky scifi guns" because we designed ourselves into a corner by making RND the one-stop-shop for virtually any technology not available at roundstart.

I've pitched ideas for moving some of RND's capabilities to Engineering and Security but it only seems to draw the most counter-productive response of "RND is such bullshit, we can't let it spread to any other department", when maybe it wouldn't be bullshit if it wasn't a one-stop-shop for powergamey shit. Every successful role offers something powerful, but its usually niche.

**Comment by Iamgoofball** [comment]
Techwebs is just a time limit that suspiciously lines up with how long it takes to get all your mats

**Comment by Iamgoofball** [comment]
@arianya 
It's another "whine to the mods to DELETE/CENSOR THIS" post

**Comment by Iamgoofball** [comment]
@KorPhaeron 
Don't fall for the R&D blameshift tactic, the problem is with guns

**Comment by Iamgoofball** [comment]
@robustin yeah we need to castrate R&D because it keeps marking everything's balance as based around it

**Comment by Iamgoofball** [comment]
@panthepusher ends up with R&D just moving to cargo, relocating the problem

**Comment by Iamgoofball** [comment]
@JohnMcLaime
Have you considered that every gamemode starts with "destroy R&D or lose the late game" is bad

**Comment by Iamgoofball** [comment]
@anonymousnow see above relocating the problem

**Comment by Iamgoofball** [comment]
Additionally @KorPhaeron 
Time gates aren't balance, they don't work well and just bore people until they alt tab and come back

It's literally pushing the problem to 30 minutes later, not a fix

**Comment by Iamgoofball** [comment]
@qbopper this problem predates your time

**Comment by Robustin** [comment]
I'll end my rant with a simple pitch: 

Let security get a budget for experimental weapons based on performance: Percentage of crew still alive, Security satisfaction surveys PDA'd to random crew every _____ minutes (bonuses would depend on the average response to these surveys, so even 50% satisfaction would be rewarded if the average is 20%), and even the power to "process" enemies of the station - holding and deporting those convicted of "capital crimes" with a special prison shuttle (with a long waiting period) that would require the brig to start acting like a brig and not just an oversized execution chamber + armory (prisoners would either be return to the station after being "rehabilitated", possibly with mindshields, and blocked from latejoin antag roles, with no/limited memory of their previous time aboard. 

It would give sec an incentive to behave well, to create an interesting balance between quick and easy solutions (harm baton, throw into sec morgue) or a potentially harder but more rewarding solution (keep prisoner alive and secure, enter criminal record, transport them to the prison ship) that would grow sec's budget for these crazy weapons.

Perhaps RND could still offer "design disks", limited to one submission per prison shuttle cycle (~15m), where a weapon design could be submitted an obtained by security for little or no budget expense, and a crate of said weapons would be delivered. 

It's a big change but it literally can't be worse than just "RND gets a monopoly on cool guns while Security remains a shit role because you get your best equipment at roundstart and all you do is slowly run out of supplies while everyone else gets cool shit for doing their job".

**Comment by praisenarsie** [comment]
i agree with point 6 of op

**Comment by Iamgoofball** [comment]
@robustin or instead of coding a massive system to handle guns just to argue for 6 months until it gets reverted, we just remove R&D guns

**Comment by Bawhoppen** [comment]
Robustin's ideas revolve around arbitrary points way too much, but at this stage, I'd still prefer it

**Comment by Spaceinaba** [comment]
you all act like you dont have firing pins in the brig and cant walk down and print your own sheilds and guns

**Review by ChangelingRain — APPROVED** [review]

**Comment by KorPhaeron** [comment]
Also worried about this because RnD is the backbone of our "jobs" that bribe players into playing minigames instead of griefing one another.

Theoretically doing RnD and working with mining to create and distribute stuff to various departments has the potential to be the most interesting "job" as far as progression goes. The research itself is very quick (surely techwebs will fix it!) and mining costs are too low across the board but removing some of the rewards doesn't fix any of that.

I'd rather the source of new weapons involve researching and building them out of mined materials rather than just generating them from thin air in cargo.

**Comment by KorPhaeron** [comment]
Also worried about this because RnD is the backbone of our "jobs" that bribe players into playing minigames instead of griefing one another.

Theoretically doing RnD and working with mining to create and distribute stuff to various departments has the potential to be the most interesting "job" as far as progression goes. The research itself is very quick (surely techwebs will fix it!) and mining costs are too low across the board but removing some of the rewards doesn't fix any of that.

I'd rather the source of new weapons involve researching and building them out of mined materials rather than just generating them from thin air in cargo.

**Comment by Grazyn** [comment]
@Iamgoofball why not make a machine that spawns in sec and is synched to RnD but only prints guns (i.e. stuff from the "weapons" section of the protolathe) and is ID locked so that science can't just build their own copy

**Comment by Iamgoofball** [comment]
See: r&d anger over not being able to print gunparts in guncrafting, r&d players get mad over being decentralized

**Comment by sawrge** [comment]
When the fuck have you ever cared about angering people aside from as an excuse for being lazy

**Comment by GuyonBroadway** [comment]
My stance is R&D is boring as shit to play and gutting the fun tools means nobody will even bother to waste the first-half of an antag round to even bother. Might as well run xenobio where your time sunk has the payoff of a nice loyal mob army. Or play robotics and watch as new recruits are hammering on the doors.

**Comment by Arzion5** [comment]
You have many ways of dealing with a ravaging horde of R&D scientists. You can always ask the AI to wipe their research, you can always ruin their research server then screwdriver the monitor for a complete research wipe, you can always just bomb it if they start attacking people, but admins might get pissed. Either way, if they're being uppity assholes, you can always just retaliate without anyone batting an eye.

**Comment by Iamgoofball** [comment]
@Arzion5 

>ask AI to wipe

"LAW 2 AI DONT WIPE R&D"
they can build ai consoles and ai upload consoles and ai law boards fyi

>break into research server

they're the heaviest armed and armored people on station with battle mecha

>bomb it

they control bombs

**Comment by GuyonBroadway** [comment]
If R&D so much as prints a plasma cutter without prior authorization its usually justification enough to round up the Gestapo and go krystalnacht on the science wing.

**Comment by Iamgoofball** [comment]
@GuyonBroadway when was the last time this happened

**Comment by GuyonBroadway** [comment]
Its not the fault of R&D that sec don't do their job and keep the station in line.

**Comment by Arzion5** [comment]
@Iamgoofball That's why you do it silently. You shouldn't ask the AI over the radio about this, you should send it a PDA message or something similar, put a paper to a camera, call a holopad and tell it, and more.

**Comment by Iamgoofball** [comment]
@Arzion5 

>pda message

they control the PDAs

additionally any smart R&D main would make sure the AI has laws against wiping R&D every time this happens, or they'll build their own AI, or they'll complain that wiping R&D is griefing everyone/human harm

**Comment by GuyonBroadway** [comment]
>additionally any smart R&D main would make sure the AI has laws against wiping R&D every time this happens

When does that ever happen?

**Comment by Iamgoofball** [comment]
as soon as they realize they can do it

**Comment by DoctoreMad** [comment]
Actually, never.

**Comment by Arzion5** [comment]
@Iamgoofball I'm pretty sure the only console with access to the PDA logs is in the Telecomms, and even then you require the password given to the CE

**Comment by Iamgoofball** [comment]
the RD has the password, idiot

**Comment by GuyonBroadway** [comment]
It ain't in the RD's locker or on his desk dummy.

**Comment by Iamgoofball** [comment]
not on all maps

**Comment by Arzion5** [comment]
The one true map that matters is Boxstation, dum-dum y

**Comment by KorPhaeron** [comment]
For better or worse RD has become the backbone of station progression/cooperation over the years which is important both for baiting people into doing their jobs and for the station keeping pace with antags that also power up as the round goes on and I'd rather see the system decentralized and balanced (mining costs, RD time, techwebs, departmental lathes) rather than just taking an axe to the various rewards

This just feels like a random stab at RD than a serious attempt to overhaul or salvage the system

**Comment by AnonymousNow** [comment]
Couldn't have put it better myself.

The notion of having things in every department that are roughly equivalent to "research" is a curious, but strong one. Warden putting time into gunsmithing. Librarian putting time into researching spells. Cargo technicians putting time into improved mining and lathing equipment. That sort of thing.

**Comment by Slignerd** [comment]
>the RD has the password, idiot

W-when's the last time goofball actually played?

**Comment by YakumoChen** [comment]
RD can't even access telecomms you stupid fuck

**Comment by Iamgoofball** [comment]
the PDA server is IN THE SAME FUCKING ROOM AS THE RD SERVER

**Comment by Repukan** [comment]
No, PDA server is in engineering

**Comment by Repukan** [comment]
CE gets the password

**Comment by Iamgoofball** [comment]
when was this change made

**Comment by Repukan** [comment]
Probably a couple weeks after you stopped playing.

**Comment by Iamgoofball** [comment]
probably

**Comment by Repukan** [comment]
So three years or so

**Comment by Iamgoofball** [comment]
i believe so

**Comment by PKPenguin321** [comment]
Literally years ago

**Pull request closed without merging**